### PR TITLE
Accept `..` in incorrect position to avoid further errors

### PR DESCRIPTION
**Pull request by estebank**
We currently give a specific message when encountering a `..` anywhere
other than the end of a pattern. Modify the parser to accept it (while
still emitting the error) so that we don't also trigger "missing fields
in pattern" errors afterwards.

Add suggestions to either remove trailing `,` or moving the `..` to the
end.

Follow up to #49268.

### CODE REVIEW

**Comment by petrochenkov** [comment]
r=me after fixing tidy

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -3667,100 +3667,163 @@ impl<'a> Parser<'a> {
...
+                ident: fieldname,
+                pat: subpat,
+                is_shorthand,
+                attrs: attrs.into(),
+           }
+        })
+    }
+
     /// Parse the fields of a struct-like pattern
+    #[allow(unused_assignments)]
```
[the remark this answers is not in the document]
Comment (estebank) [reply src/libsyntax/parse/parser.rs]: Leftover from a false positive in a previous iteration of the code.

File: src/libsyntax/parse/parser.rs [hunk 2]
```diff
@@ -3667,100 +3667,163 @@ impl<'a> Parser<'a> {
...
+            fields.push(match self.parse_pat_field(lo, attrs) {
+                Ok(field) => field,
+                Err(err) => {
+                    if let Some(mut delayed_err) = delayed_err {
+                        delayed_err.emit();
+                    }
+                    return Err(err);
+                }
             });
+            ate_comma = self.eat(&token::Comma);
```
[the remark this answers is not in the document]
Comment (estebank) [reply src/libsyntax/parse/parser.rs]: I made a cursory test and the behavior when encountering multiple commas hasn't changed, the parser fails and no attempt at recovery is made.

**Pull request merged**